### DISCUSSION TITLE
Vike + a different bundler other than Vite? (Farm, RsPack)

### DISCUSSION THREAD

**Discussion by Blankeos** [vikejs/vike#1737] [Discussion]
[Not really a bug report, just opening an interesting discussion for 🥼 Science 🧪]

These past few weeks, some new faster bundlers than Vite have been popping up: RsPack https://www.rspack.dev/ (familiar Webpack config) and Farm https://www.farmfe.org/ (familiar Vite config). 

**Farm** is probably the most attractive with Vite codebases (it does try to be backwards compatible with Vite). I have tried it as a drop-in replacement with Vike [but no luck](https://github.com/orgs/farm-fe/discussions/1570#discussion-6916574). Considering Vike is very coupled with Vite, I can see that it might be a challenge.

Granted, Vite's already fast enough so there's probably no motivation to do so. But for science :man_scientist: , I'm wondering how much of a work effort it is to probably make Vike compatible with **Farm**? 

I'm wondering if it has something to do with the Vite SSR -> Farm SSR configuration: https://www.farmfe.org/docs/migration/from-vite

**Comment by brillout** [comment]
Hi Carlo. Actually, the *vast* majority of Vike's logic is Vite agnostic (more than 99%). So, in principle, making Vike bundler agnostic is much easier than it seems. That said, there is still a lot of glue code so it's a significant endeavour.

Vite is far superior than any alternative I'm aware of. (Performance is only a small fraction of what makes Vite uniquely great.) So, yea, while it's possible to make Vike bundler agnostic it clearly isn't a priority for now.

Btw. thanks for helping folks on Discord, it's much appreciated :green_heart:. (Speaking of which, I'm thinking we should have a `vike.dev/getting-started` that introduces key concepts like SSR and SSG for newcomers to frontend development. There are a lot of redundant questions about the overall architecture of frontend apps.)
